### PR TITLE
feat: custom fields component to registration form

### PR DESCRIPTION
**Pull request by hugocostadev**
## Proposed changes (including videos or screenshots)

Since the `CustomFieldsForm` is used across pages I leverage the component to `ui-client` package to be used anywhere and edited smalls parts to adapt to registration page and UsersInfo page

Changes:
- [x]  moved CustomFieldV2 to packages/ui-client
- [x] moved `useAccountsCustomFields.ts` to `packages/ui-context`
- [x] rename omnichannel `CustomFieldsForm.js` to `NewCustomFieldsForm.js` to avoid confusion btw names
- [x]  ✅ Add the `CustomFieldForm` to the `NewCustomFieldsForm` page
- [x]  ✅ Add the `CustomFieldForm` to the registration page
- [x]  ✅ Add the `CustomFieldForm` to the Users Table actions (add and edit)
- [x]  ✅ Add the `CustomFieldForm` to the `RegisterUsername.tsx` page when an anonymous user wants to send a message
- [x] ✅ Add custom Fields validation and registration to `users.register` endpoint 
- [x]  🎨 Styled buttons from UsersTable actions (add and edit)
- [x]  ❌ **delete** `AccountsCustomFieldsAssembler`
- [x]  ❌ **delete** `CustomFieldV2` from omnichannel
- [x]  ❌ **delete** `CustomField` from meteor/components
- [ ]  🔬 Write **Tests** to custom fields forms

## Further comments

[SUP-196]

[SUP-196]: https://rocketchat.atlassian.net/browse/SUP-196?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by rique223 — DISMISSED** [review]
LGTM!

File: packages/ui-contexts/src/types/CustomFieldMetadata.ts
Comment (dougfabris) [inline packages/ui-contexts/src/types/CustomFieldMetadata.ts:1]: Shouldn't this types live on `@rocket.chat/core-typings`?

File: apps/meteor/client/views/admin/users/EditUser.js [hunk 1]
```diff
@@ -131,10 +131,10 @@ function EditUser({ data, roles, onReload, ...props }) {
 				<Field.Row>
 					<Box display='flex' flexDirection='row' justifyContent='space-between' w='full'>
 						<Margins inlineEnd='x4'>
-							<Button flexGrow={1} type='reset' disabled={!canSaveOrReset} onClick={reset}>
+							<Button danger flexGrow={1} type='reset' disabled={!canSaveOrReset} onClick={reset}>
```
Comment (dougfabris) [inline apps/meteor/client/views/admin/users/EditUser.js:134]: Let's avoid the usage of danger variation for reset action

**Review by dougfabris — CHANGES_REQUESTED** [review]

File: packages/ui-contexts/src/types/CustomFieldMetadata.ts
Comment (hugocostadev) [reply packages/ui-contexts/src/types/CustomFieldMetadata.ts:1]: Maybe ...🤔 I'll check it here, thanks

File: apps/meteor/client/views/admin/users/EditUser.js (hunk 1 above)
Comment (hugocostadev) [reply apps/meteor/client/views/admin/users/EditUser.js:134]: Got it, I'll fix it, thanks

File: packages/ui-client/src/components/CustomFieldsForm.tsx
Comment (dougfabris) [inline packages/ui-client/src/components/CustomFieldsForm.tsx:1]: what do you think about wrapping the `Fields` with `FieldsGroup` to get more space between them? 👀

Comment (hugocostadev) [reply packages/ui-client/src/components/CustomFieldsForm.tsx:1]: Done! Thanks

**Review by rique223 — APPROVED** [review]

**Review by ggazzo — APPROVED** [review]

**Review by dougfabris — DISMISSED** [review]

**Review by tapiarafael — APPROVED** [review]

**Review by rique223 — DISMISSED** [review]

**Review by tapiarafael — APPROVED** [review]

**Review by tapiarafael — DISMISSED** [review]

**Review by ggazzo — DISMISSED** [review]

**Review by ggazzo — APPROVED** [review]

**Pull request merged**